engines/zypper_auth/lib/zypper_auth/engine.rb

```diff
@@ -31,31 +31,14 @@ def verify_instance(request, logger, system)
       )
 
       is_valid = verification_provider.instance_valid?
+      return false if is_valid && system.hybrid? && !handle_scc_subscription(request, system, verification_provider, logger)
```

Comment:
a `BYOS` system will raise an exception and will not reach line 34 (that's the behavior before this change)

Reply:
Alright we depend on a ` raise InstanceVerification::Exception` in the `instance_valid?` method inside the CSP specific verification code. It would probably be good to add a comment here. This is behavior a 3rd party implementer of a CSP would have to duplicate.

An explanation like this would sure have been nice, but anyway, I can certainly look things up with enough pointers.